src/main/java/com/google/sps/servlets/EventServlet.java

```diff
@@ -65,7 +73,54 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
     @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        // ToDo: Mingyi implment this method.
-        response.getWriter().println("You've called the post request");
+        String jsonBody = new BufferedReader(new InputStreamReader(request.getInputStream())).lines().collect(
+            Collectors.joining("\n"));
+
+        // Error handling: return error if jsonBody is empty
+        if (jsonBody == null || jsonBody.trim().length() == 0) {
+            return;
+        }
+
+        // Parse the jsonBody String to a Gson object
+        Gson gson = new Gson();
+        Type type = new TypeToken<Map<String, String>>(){}.getType();
+        Map<String,String> myMap = gson.fromJson(jsonBody, type);
+        
+        // Get the value entered in the form  
+        String event_Lat = myMap.get("lat");
+        Double event_Lat_D = Double.parseDouble(event_Lat);
+        String event_Lng = myMap.get("lng");
+        Double event_Lng_D = Double.parseDouble(event_Lng);
+        String event_name = myMap.get("name");
+        String event_desc = myMap.get("description");
+        String event_loc = myMap.get("loc");
+        String event_time = myMap.get("date");
```

Comment:
Could you change this line to "TimestampValue event_time = TimestampValue.of(Timestamp.parseTimestamp((myMap.get("date"))));"?

I thought about changing the type of the date property in the datastore to String but I think it would be easier to manage the data if it's stored as Date and Time. You'll need to import import com.google.cloud.Timestamp and import com.google.cloud.datastore.TimestampValue.